src/System.Private.CoreLib/shared/System/Globalization/CharUnicodeInfoData.cs

```diff
@@ -2,1141 +2,1466 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
-
 namespace System.Globalization
 {
     public static partial class CharUnicodeInfo
     {
-        // THE FOLLOWING DATA IS AUTO GENERATED BY GenUnicodeProp.pl SCRIPT UNDER THE TOOLS FOLDER
+        // THE FOLLOWING DATA IS AUTO GENERATED BY GenUnicodeProp program UNDER THE TOOLS FOLDER
         // PLEASE DON'T MODIFY BY HAND
 
 
-        // 12:4:4 index table of the Unicode category data.
-        private static ushort[] s_pCategoryLevel1Index = new ushort[]
+        // 11:5:4 index table of the Unicode category data.
+        private static ReadOnlySpan<byte> CategoryLevel1Index => new byte[2176]
```

Comment:
I think this relies on quite advanced JIT optimization to work better than previous code. Basically we are generating IL code which we hope to be executed differently than it's written. It's using C# getter syntax with newarr and expects JIT to convert this to aloc free read from static memory map.

/cc @jkotas @vargaz

Reply:
The problem seems to be that this code will still compile with 
csc /langversion:7.1, but it will generate a newobj + InitializeArray pair for _every_ access to the
variable.

Reply:
Presumably, the C# codegen should not be language version specific but based on `System.ReadOnlySpan`1<uint8>::.ctor(void*,int32)` availability

Reply:
Right, this does not depend on any special JIT optimizations. This optimization is done by Roslyn: https://github.com/dotnet/roslyn/pull/24621.

Reply:
> I'd suggest roslyn should fail to compile this code when it cannot make the optimization

This is valid code regardless of whether it can be optimized.  Using a new enough compiler, which the coreclr repo does, is sufficient to guarantee it be optimized.  I agree it's subtle, but that's how the feature is exposed by the compiler.

Reply:
> I'd suggest roslyn should fail to compile this code when it cannot make the optimization

Open issue on this in Roslyn repo if you would like to see this happen.

It will never happen in practice since `System.ReadOnlySpan``1<uint8>::.ctor(void*,int32)` is present in all `ReadOnlySpan` implementations out there.

Reply:
It'll actually happen in practice because by default csc defaults to langversion:7 which generates the ugly code with newarr not the optimized version.

Reply:
@marek-safar, what are you arguing for?  That we not take advantage of improved C# compiler support to improve the efficiency of coreclr?

Reply:
As I suggested above I think this feature should be enabled not based on /langversion but based on `System.ReadOnlySpan``1<uint8>::.ctor(void*,int32)` availability

Reply:
> As I suggested above I think this feature should be enabled not based on /langversion but based on System.ReadOnlySpan``1<uint8>::.ctor(void*,int32) availability

Ok, thanks. Sounds like an issue for the dotnet/roslyn repo.